perf_guide.md

```diff
@@ -67,17 +67,49 @@ addition is 2.7x slower than the one using vertical vector operations!
 
 ## Target-features
 
+Not all processors of a certain architecture will have SIMD processing units,
+and using a SIMD instruction when it's not supported will trigger a CPU exception.
+
+To allow building safe, portable programs, the Rust compiler will **not**, by default,
+generate any sort of vector instructions, unless it can statically determine
+they are supported (for example, on AMD64, SSE2 support is architecturally guaranteed).
```

Comment:
I'll expand here a bit, since this does not only depend on `x86_64`, but on the whole target triple. For example, `x86_64-unknown-linux-gnu` enables up to `sse2`, but `x86_64-apple-darwin` enables up to `ssse3`.  I'd also add that the target features that each target always supports is specified in the [`librustc_target` crate](https://github.com/rust-lang/rust/tree/master/src/librustc_target/spec).

Reply:
Expanding on how to find this information, all targets in the `librustc_target` crate define a `base_cpu`, which enables most target features, but some targets also add target features on top of the base cpu. The features that each cpu enables are specified in LLVM target table gen files:

* [lib/Target/X86/X86.td](https://github.com/llvm-mirror/llvm/blob/038d08360ce46e07a5a48dc953f6157cac194cad/lib/Target/X86/X86.td)
* [lib/Target/AArch64/AArch64.td](https://github.com/llvm-mirror/llvm/blob/038d08360ce46e07a5a48dc953f6157cac194cad/lib/Target/AArch64/AArch64.td)